lib/web/apiserver.go

```diff
@@ -2678,9 +2679,12 @@ func makeUnifiedResourceRequest(r *http.Request) (*proto.ListUnifiedResourcesReq
 		PinnedOnly:          values.Get("pinnedOnly") == "true",
 		PredicateExpression: values.Get("query"),
 		SearchKeywords:      client.ParseSearchKeywords(values.Get("search"), ' '),
-		UseSearchAsRoles:    values.Get("searchAsRoles") == "yes",
-		IncludeLogins:       true,
-		IncludeRequestable:  values.Get("includeRequestable") == "true",
+		// includeRequestable requires a searchAsRoles request, but we set it here instead of the frontend
+		// to protect the frontend from accidentally requesting a "SearchAsRoles" request to older proxy versions
+		// and then returning a bunch of resources that won't include "Requires Request" flags.
+		UseSearchAsRoles:   values.Get("searchAsRoles") == "yes" || includeRequestable,
+		IncludeLogins:      true,
+		IncludeRequestable: includeRequestable,
```

Comment:
Oh, it's probably too late but I feel that more clear API would be:
`ShowResources: 'only-accessible' | 'only-requestable' | 'all'`;

Then I'd deprecate `UseSearchAsRoles`.
Now the client (like Connect) can pass `UseSearchAsRoles: false` and `IncludeRequestable: true`, and I don't what would be the result of this :)

Reply:
Hmmm, I think it could be a bit cleaner. `only-x` will still be performant because we know that we can either use the base auth context or the extended auth context, and no need to do two RBAC checks. I will see if I can make this change without too much breakage

Reply:
If you decide to change it, I'd think about a better name, `ShowResources` isn't particularly descriptive.
Maybe `AccessLevel: 'only-available' | 'only-requestable' | 'all'` would be better.